src/BookReader.js

```diff
@@ -267,6 +267,32 @@ BookReader.prototype.setup = function(options) {
   });
 };
 
+/**
+ * Get all the HTML Elements that are being/can be rendered.
+ * Includes cached elements which might be rendered again.
+ */
+BookReader.prototype.getActivePageContainerElements = function() {
+  let containerEls = Object.values(this._modes.mode2Up.pageContainers).map(pc => pc.$container[0]);
+  if (this.mode != this.constMode2up) {
+    containerEls = containerEls.concat(this.$('.BRpagecontainer').toArray());
+  }
+  return containerEls;
+};
+
+/**
+ * Get the HTML Elements for the rendered page. Note there can be more than one, since
+ * (at least as of writing) different modes can maintain different caches.
+ * @param {PageIndex} pageIndex
+ */
+BookReader.prototype.getActivePageContainerElementsForIndex = function(pageIndex) {
+  const mode2UpContainer = this._modes.mode2Up.pageContainers[pageIndex]?.$container?.[0];
+  let containerEls = mode2UpContainer ? [mode2UpContainer] : [];
+  if (this.mode != this.constMode2up) {
+    containerEls = containerEls.concat(this.$(`.pagediv${pageIndex}`).toArray());
```

Comment:
q: what happens when 2up container exists && 1up/thumb containers exist for the same index.. will there be a possible drift between views or will it not matter? if the latter, where in the code does it get refreshed?

Reply:
Currently, Mode2Up is the only mode that caches PageContainers. So even when it's not enabled, it could still have PageContainers that need to be updated. Mode1Up/ModeThumb only have PageContainers for what's currently rendered into the HTML; so when they're not enabled, they don't having anything (I believe). Does that answer your question?

Longer term we might want all the modes sharing the same PageContainer cache, but for now each has its own.